docs/sources/developer/architecture/initialization.md

```diff
@@ -0,0 +1,85 @@
+# Initialization process
+
+The initialization process is pretty complex as it has to deal with a lot of components. Below is a step by step
+description of how it is initialized under the hood.
+
+Please refer to the components documentation for more details about each component.
+
+## Core package
+
+### 1. Unpatched Console
+
+The first component that is initialized is the unpatched console. This is needed because it is needed by the internal
+logger component. By default, the unpatched console is initialized with the global `console` object but it can be
+overwritten by the user if is provided in the config object.
+
+### 2. Internal Logger
+
+The internal logger is initialized immediate after the unpatched console but before anything else happens because it
+is needed to print out debug messages. The internal logger relies on the unpatched console and not on the `console`
+object because the latter can be overwritten by instrumentations and we don't want to send those debug messages as
+signals.
+
+### Isolation check
+
+Before any other component is initialized, we run an isolation check.
+
+By default, Faro is a global singleton that can be initialized only once. However, multiple instances can be initialized
+by marking them as isolated using the `isolate` config option. This is useful in large applications that contain
+multiple sub-applications which should be instrumented independently.
+
+The process of isolation check is pretty easy: when a singleton Faro instance is initialized, we define it on the
+global object of the environment (either `window` in browsers or `global` in Node.js). If we detect that there is an
+instance already initialized, we stop the process here.
+
+However, the isolation mode should not be mistaken with global object exposure. The global object exposure is a way to
+make Faro available to the end-user on the global object. Two Faro instances can be instantiated as isolated but they
+can still be exposed on the global object independently of each other.
+
+### 3. Metas
+
+The metas are the next component that is initialized as they are required by all other subsequent components. The API
+requires it because whenever a signal is sent, the immediate meta values are added to it in order to keep them in sync.
+The instrumentations and transports require the metas for exposing them to the developers that are writing their own
+instrumentations and transports.
+
+However, the default metas values are not registered now (check the Initial values section below for more details).
+
+### 4. Transports
+
+The next component that is initialized is the transports components. It comes before the API component because the API
```

Comment:
"the transports component" (remove "s" from components)?

Reply:
I prefer to keep it like this since the entire mechanism for transports is referred to with the plural everywhere.